go/runtime/history/db.go

```diff
@@ -270,6 +270,34 @@ func (d *DB) getLastBlock() (*roothash.AnnotatedBlock, error) {
 	return &blk, nil
 }
 
+func (d *DB) deleteBlock(round uint64) error {
+	return d.db.Update(func(tx *badger.Txn) error {
+		it := tx.NewIterator(badger.IteratorOptions{
+			Prefix: blockKeyFmt.Encode(),
```

Comment:
This should not work. You should pass `round` and check if `it.Item()` returns the right key. 

Why don't you just do `return tx.Delete(blockKeyFmt.Encode(round))`?